pkg/cmd/generate-cgo/main.go

```diff
@@ -0,0 +1,197 @@
+// Copyright 2023 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"text/template"
+
+	"github.com/cockroachdb/errors"
+
+	"github.com/alessio/shellescape"
+)
+
+type configuration struct {
+	Os   string
+	Arch string
+}
+
+var (
+	workspace                  string
+	bazelBin                   string
+	archivedCdepConfigurations = []configuration{
+		{"linux", "amd64"},
+		{"linux", "arm64"},
+		{"darwin", "amd64"},
+		{"darwin", "arm64"},
+		{"windows", "amd64"},
+	}
+)
+
+func init() {
+	flag.StringVar(&workspace, "workspace", "", "workspace path")
+	flag.StringVar(&bazelBin, "bazel-bin", "", "path to bazel/bin")
+}
+
+func logCommand(cmd string, args ...string) {
+	var fullArgs []string
+	fullArgs = append(fullArgs, cmd)
+	fullArgs = append(fullArgs, args...)
+	log.Printf("$ %s", shellescape.QuoteCommand(fullArgs))
+}
+
+// getArchivedCdepString returns a non-empty string iff the force_build_cdeps
+// config is not being used. This string is the name of the cross config used to
+// build the pre-built c-deps, minus the "cross" prefix. This can be used to
+// locate the pre-built c-dep in
+// $EXECUTION_ROOT/external/archived_cdep_{LIB}_{ARCHIVED_CDEP_STRING}.
+// If the returned string is empty then force_build_cdeps is set in which case
+// the (non-pre-built) libraries can be found in $BAZEL_BIN/c-deps/{LIB}_foreign.
+//
+// You MUST build //build/bazelutil:test_force_build_cdeps before calling this
+// function.
+func getArchivedCdepString(bazelBin string) (string, error) {
+	var ret string
+	// If force_build_cdeps is set then the prebuilt libraries won't be in
+	// the archived location anyway.
+	forceBuildCdeps, err := os.ReadFile(filepath.Join(bazelBin, "build", "bazelutil", "test_force_build_cdeps.txt"))
+	if err != nil {
+		return "", err
+	}
+	// force_build_cdeps is activated if the length of this file is not 0.
+	if len(forceBuildCdeps) == 0 {
+		for _, config := range archivedCdepConfigurations {
+			if config.Os == runtime.GOOS && config.Arch == runtime.GOARCH {
+				ret = config.Os
+				if ret == "darwin" {
+					ret = "macos"
+				}
+				if config.Arch == "arm64" {
+					ret += "arm"
+				}
+				break
+			}
+		}
+	}
+	return ret, nil
+}
+
+func getExecutionRoot() (string, error) {
+	args := []string{"info", "execution_root", "--color=no"}
```

Comment:
Doesn't really make much sense that `workspace` and `bazel-bin` have to be passed in, but `execution_root` does not. I would say just have `generate-cgo` look all of them up by itself.